policies/redis/consumer/streams.go

```diff
@@ -98,6 +145,20 @@ func (es eventStore) handleAgentGroupRemove(ctx context.Context, groupID string,
 	return nil
 }
 
+func (es eventStore) handleSinkRemove(ctx context.Context, sinkID string, token string) error {
+
```

Comment:
Maybe first you should delete the sink from the dataset, after that, you should retrieve the dataset and check the size of the sinks it equals zero, and then you inactive the dataset. What do you think?